### PR TITLE
[SYCL][CI] Update nightly to use resolve_matrix properly

### PR DESCRIPTION
**Pull request by pvchupin**
Change https://github.com/intel/llvm/pull/6352 done earlier introduced
regression to SYCL nightly workflow. This change is to fix it.

### CODE REVIEW

**Comment by pvchupin** [comment]
Opaque pointers testing failed before. Not related to the change.

**Review by bader — APPROVED** [review]
LGTM, although I would give another name to this job/script.
"Resolve matrix" sounds unclear to me and I had to look into script sources to understand what it's doing.

**Comment by pvchupin** [comment]
Agree. I just keep using name introduced earlier. Will update with a separate NFC PR.

File: .github/workflows/sycl_nightly.yml [hunk 1]
```diff
@@ -10,24 +10,30 @@ on:
...
+  resolve_matrix:
+    name: Resolve Test Matrix
+    uses: ./.github/workflows/sycl_resolve_test_matrix.yml
+    with:
+      lts_config: "ocl_gen9;ocl_x64"
+
   ubuntu2004_build_test:
     if: github.repository == 'intel/llvm'
     uses: ./.github/workflows/sycl_linux_build_and_test.yml
+    needs: resolve_matrix
```
Comment (aelovikov-intel) [inline .github/workflows/sycl_nightly.yml:22]: I think this stopped running SYCL end-to-end tests due to missing

```
with:
  lts_matrix: needs.resolve_matrix.outputs.lts_matrix
```

@bader , do we need to fix that? Or should we just remove matrix generation from nightly alltogether?

Comment (bader) [reply .github/workflows/sycl_nightly.yml:22]: I'll let @sergey-semenov to decide on the runtime library validation requirements/process.

Comment (aelovikov-intel) [reply .github/workflows/sycl_nightly.yml:22]: @stdale-intel and I want to use all the E2E tests as the criteria for "latest-good" nightly release, so I'm re-enabling missing tasks in https://github.com/intel/llvm/pull/10254.

**Pull request merged**